### PR TITLE
Don't exit insight edit mode on Esc click

### LINKED ISSUE #7857: Pressing ESC analyzing an insight exits editing it
## Bug description

Currently, when I press X analyzing a _new_ insight it exits the edit mode and only shows me the graph. This is annoying already.

If I have the quick action menu open at the same time it becomes even more so - if I try to exit the menu via ESC it exits both the menu as well as editing the insight.

## Expected behavior

Drop the ESC behavior on insights - I don't see it being useful.

## How to reproduce

1. Click + to add insight
2. Ctrl + K to open menu (optional)
3. Press ESC
4. See editing is stopped

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

## Additional context

cc @Twixes and @mariusandra for prioritization

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Small change to resolve #7857.

### CODE REVIEW

**Review by macobo — APPROVED** [review]
:+1: Code-wise, good to get a second opinion product-wise though :)

**Comment by Twixes** [comment]
Yup, pinged Paolo here for that reason.

File: frontend/src/scenes/insights/Insight.tsx [hunk 1]
```diff
@@ -86,11 +86,6 @@ export function Insight({ shortId }: { shortId?: InsightShortId } = {}): JSX.Ele
         l: {
             action: () => handleHotkeyNavigation(InsightType.LIFECYCLE, 'l'),
         },
-        escape: {
-            // Exit edit mode with Esc. Full screen mode is also exited with Esc, but this behavior is native to the browser.
-            action: () => setInsightMode(ItemMode.View, InsightEventSource.Hotkey),
-            disabled: insightMode !== ItemMode.Edit,
-        },
         e: {
             action: () => setInsightMode(ItemMode.Edit, InsightEventSource.Hotkey),
```
Comment (paolodamico) [inline frontend/src/scenes/insights/Insight.tsx:90]:
```suggestion
            action: () => setInsightMode(insightMode === ItemMode.Edit ? ItemMode.View : ItemMode.Edit, InsightEventSource.Hotkey),
```

File: frontend/src/scenes/insights/Insight.tsx [hunk 2]
```diff
@@ -86,11 +86,6 @@ export function Insight({ shortId }: { shortId?: InsightShortId } = {}): JSX.Ele
...
             disabled: insightMode !== ItemMode.View,
```
Comment (paolodamico) [inline frontend/src/scenes/insights/Insight.tsx:91]:
```suggestion
```

**Review by paolodamico — APPROVED** [review]
I think the escape behavior is useful when you're just browsing existing insights, but I can understand how it can be confusing. Suggested an improvement.

File: frontend/src/scenes/insights/Insight.tsx (hunk 1 above)
Comment (Twixes) [reply frontend/src/scenes/insights/Insight.tsx:90]: I tried this out, but it's quite a confusing action to exit edit mode having neither discarded changes nor saved them – the results in view mode don't match up with what's saved then. For now I'll opt for the initial solution of just disabling Escape.

Comment (paolodamico) [reply frontend/src/scenes/insights/Insight.tsx:90]: Oh you're right that makes sense, thanks @Twixes

**Pull request merged**